opal/mca/accelerator/accelerator.h

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2014-2021 Intel, Inc. All rights reserved.
+ * Copyright (c) 2015      Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2017-2022 Amazon.com, Inc. or its affiliates.
+ *                         All Rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#ifndef OPAL_ACCELERATOR_H
+#define OPAL_ACCELERATOR_H
+
+#include "opal/types.h"
+
+#include "opal/mca/mca.h"
+
+BEGIN_C_DECLS
+
+/**
+ * Accelerator flags
+ */
+#define MCA_ACCELERATOR_UNIFIED_MEMORY 0x00000001
+
+/**
+ * Transfer types.
+ * UNSPEC - Not specified
+ * HTOH - Host to Host
+ * HTOD - Host to Device
+ * DTOH - Device to Host
+ * DTOD - Device to Device
+ */
+typedef enum {
+    MCA_ACCELERATOR_TRANSFER_UNSPEC = 0,
+    MCA_ACCELERATOR_TRANSFER_HTOH,
+    MCA_ACCELERATOR_TRANSFER_HTOD,
+    MCA_ACCELERATOR_TRANSFER_DTOH,
+    MCA_ACCELERATOR_TRANSFER_DTOD,
+} opal_accelerator_transfer_type_t;
+
+/**
+ * Check whether a pointer belongs to an accelerator or not.
+ * interfaces
+ *
+ * @param[IN] addr           Pointer to check
+ * @param[OUT] dev_id        Returns the device id against which the memory was allocated
+ * @param[OUT] flags         May be set to indicate additional information
+ *                           about the corresponding pointer such
+ *                           as whether it belongs to unified memory.
+ *
+ * @return <0                An error has occurred.
```

Comment:
This isn't proper doxygen format.  `@retval` specifies a return value and description.  `@return` is a paragraph form describing the return value.

Does this return OPAL error codes or some other error scheme?

Reply:
verified Doxygen comments have been updated.